### ISSUE TITLE
Error + NPC disappear after a server restart

### ISSUE THREAD

**Issue by Ragehorror** [CitizensDev/Citizens#887]
Hello,
- Craftbukkit-1.1-R4
- Citizens 1.1.6
- bPermission 2.9.0

I have tried to install citizens, the first time i use the normal bukkit permission, didnt go well, Non ops could trade with a Trader or teleport with a wizard. i tough i was a mistake i made, after read a lot of issues i begon to think it was a permission error so.
# i installed bPermission with a the hope the problem would be fix now, but in the opposite it gives a new...

174 recipes
27 achievements
12:44:52 [INFO] Starting minecraft server version 1.1
12:44:52 [INFO] Loading properties
12:44:52 [INFO] Starting Minecraft server on _:45565
12:44:52 [WARNING] *_*\* SERVER IS RUNNING IN OFFLINE/INSECURE MODE!
12:44:52 [WARNING] The server will make no attempt to authenticate usernames. Be
ware.
12:44:52 [WARNING] While this makes the game possible to play without internet a
ccess, it also opens up the ability for hackers to connect with any username the
y choose.
12:44:52 [WARNING] To change this, set "online-mode" to "true" in the server.pro
perties file.
12:44:52 [INFO] This server is running CraftBukkit version git-Bukkit-1.1-R4-b19
38jnks (MC: 1.1) (Implementing API version 1.1-R4)
12:44:52 [INFO] [Safe Creeper] Loading Safe Creeper v0.4.
12:44:52 [INFO] [Safe Creeper] Safe Creeper V0.4 Enabled
12:44:52 [INFO] [BKCommonLib] Loading BKCommonLib v1.08.
12:44:52 [INFO] BKCommonLib version 1.08 enabled!
12:44:52 [INFO] [BloodMoon] Loading BloodMoon v0.12.
12:44:52 [INFO] BloodMoon 0.12: Enabled.
12:44:52 [INFO] Preparing level "Channel1"
12:44:52 [INFO] Default game type: 0
12:44:52 [INFO] Preparing start region for level 0 (Seed: 1552206508537009132)
12:44:53 [INFO] Preparing start region for level 1 (Seed: -1883299767546932141)
12:44:53 [INFO] Preparing spawn area: 65%
12:44:53 [INFO] Preparing start region for level 2 (Seed: 1552206508537009132)
12:44:54 [INFO] [bPermissions] Loading bPermissions v2.9.0.
12:44:54 [INFO] [bPermissions] Loading world: Channel1
12:44:54 [INFO] [bPermissions] Loading world: Channel1_nether
12:44:54 [INFO] [bPermissions] Loading world: Channel1_the_end
12:44:54 [INFO] [bPermissions] Enabled
12:44:54 [INFO] [AlphaChest] Loading AlphaChest v1.10.0.
12:44:54 [WARNING] Could not save config.yml to plugins\AlphaChest\config.yml be
cause config.yml already exists.
12:44:54 [INFO] [AlphaChest] loaded 0 chests
12:44:54 [INFO] [AlphaChest] version [1.10.0] enabled
12:44:54 [INFO] [ScheduledAnnouncer] Loading ScheduledAnnouncer v1.8.
12:44:54 [INFO] ScheduledAnnouncer v1.8 is enabled!

12:44:54 [INFO] [iConomy] Loading iConomy v6.0.8b.
12:44:54 [INFO] [iConomy - Celty] Enabled (14 ms)
12:44:54 [INFO] [WorldSaver] Loading WorldSaver v1.4.
12:44:54 [INFO] [WorldSaver] enabled. Visit goo.gl/HWbai for plugin info
12:44:54 [INFO] [NoWeather] Loading NoWeather v1.7.3.
12:44:54 [INFO] [Skillz] Loading Skillz v5.51.
12:44:54 [INFO] [Skillz] Metrics loaded!
12:44:55 [WARNING] [Skillz] Vault not found. Money reward -> 0
12:44:55 [INFO] [Skillz] Loaded HighScore archery, ragehorror, 3!
12:44:55 [INFO] [Skillz] Loaded HighScore acrobatics, ragehorror, 4!
12:44:55 [INFO] [Skillz] Loaded HighScore unarmed, ShadoSword, 2!
12:44:55 [INFO] [Skillz] Loaded HighScore digging, Robbinvt, 1!
12:44:55 [INFO] [Skillz]  - V5.51 Enabled!
12:44:55 [INFO] [NoAFK] Loading NoAFK v0.2.4.
12:44:55 [INFO] [NoAFK] Permission system not detected, defaulting to OP
12:44:55 [INFO] [NoAFK] version 0.2.4 enabled
12:44:55 [INFO] [MobHealth] Loading MobHealth v3.6.
12:44:55 [INFO] [MobHealth] Version 3.6 starting.
12:44:55 [INFO] [MobHealth] Permissions Disabled.
12:44:55 [INFO] [MobHealth] Spout Enabled.
12:44:55 [INFO] [MobHealth] Player Notifications Enabled.
12:44:55 [INFO] [MobHealth] Monster Notifications Enabled.
12:44:55 [INFO] [MobHealth] Animals Notifications Enabled.
12:44:55 [INFO] [MobHealth] Online.
12:44:55 [INFO] [dynmap] Loading dynmap v0.32-884.
12:44:55 [INFO] [dynmap] Using Bukkit Permissions (superperms) for access contro
l
12:44:55 [INFO] [dynmap] Loaded 15 shaders.
12:44:55 [INFO] [dynmap] Loaded 82 perspectives.
12:44:55 [INFO] [dynmap] Loaded 12 lightings.
12:44:55 [INFO] [dynmap] version 0.32-884 is enabled - core version 0.32-83
12:44:55 [INFO] [dynmap] Loaded 3 maps of world 'Channel1'.
12:44:55 [INFO] [dynmap] Loaded 2 maps of world 'Channel1_nether'.
12:44:55 [INFO] [dynmap] Loaded 2 maps of world 'Channel1_the_end'.
12:44:55 [WARNING] [dynmap] playermove trigger enabled - this trigger can cause
excessive tile updating: use with caution
12:44:55 [INFO] [dynmap] Enabled
12:44:55 [INFO] [Citizens] Loading Citizens v1.1.6.
12:44:55 [INFO] [Citizens] NPC types loaded: blacksmith, guard, healer, quester,
 trader, wizard
12:44:56 [SEVERE] Error occurred while enabling Citizens v1.1.6 (Is it up to dat
e?): de/bananaco/permissions/Permissions
java.lang.NoClassDefFoundError: de/bananaco/permissions/Permissions
        at net.citizensnpcs.permissions.BPermissionsProvider.<init>(BPermissions
Provider.java:14)
        at net.citizensnpcs.permissions.PermissionManager.<init>(PermissionManag
er.java:40)
        at net.citizensnpcs.Citizens.onEnable(Citizens.java:127)
        at org.bukkit.plugin.java.JavaPlugin.setEnabled(JavaPlugin.java:230)
        at org.bukkit.plugin.java.JavaPluginLoader.enablePlugin(JavaPluginLoader
.java:1057)
        at org.bukkit.plugin.SimplePluginManager.enablePlugin(SimplePluginManage
r.java:380)
        at org.bukkit.craftbukkit.CraftServer.loadPlugin(CraftServer.java:199)
        at org.bukkit.craftbukkit.CraftServer.enablePlugins(CraftServer.java:182
)
        at net.minecraft.server.MinecraftServer.t(MinecraftServer.java:357)
        at net.minecraft.server.MinecraftServer.a(MinecraftServer.java:344)
        at net.minecraft.server.MinecraftServer.init(MinecraftServer.java:175)
        at net.minecraft.server.MinecraftServer.run(MinecraftServer.java:408)
        at net.minecraft.server.ThreadServerApplication.run(SourceFile:465)
Caused by: java.lang.ClassNotFoundException: de.bananaco.permissions.Permissions

```
    at java.net.URLClassLoader$1.run(Unknown Source)
    at java.security.AccessController.doPrivileged(Native Method)
    at java.net.URLClassLoader.findClass(Unknown Source)
    at org.bukkit.plugin.java.PluginClassLoader.findClass(PluginClassLoader.
```

java:41)
        at org.bukkit.plugin.java.PluginClassLoader.findClass(PluginClassLoader.
java:29)
        at java.lang.ClassLoader.loadClass(Unknown Source)
        at java.lang.ClassLoader.loadClass(Unknown Source)
        ... 13 more
12:44:56 [INFO] [Citizens] An error has occurred, please wait while it is sent t
o the developers...
12:44:56 [INFO] [Citizens] Economy plugin found (iConomy v6)
12:44:56 [INFO] [Jobs] Loading Jobs v2.6.3.
12:44:56 [INFO] [Citizens] An error occurred whilst sending your error report.
12:44:56 [INFO] [Citizens] Below is the stacktrace that has been transmitted to
the developers, no other information has been included other than Citizens versi
on information.
12:44:58 [INFO] [Jobs v2.6.3] has been enabled succesfully.
12:44:58 [INFO] [Jobs] Successfully linked with iConomy 6.
12:44:58 [INFO] [MobBounty] Loading MobBounty v4.2.
12:44:58 [INFO] [MobBounty] v4.2 (Oxygen) enabled.
12:44:58 [INFO] [MobBounty] Developed by: [Mattera, Steven].
12:44:58 [INFO] [MobBounty] Config loaded.
12:44:58 [INFO] [MobBounty] Payment plugin iConomy v6 found!
12:44:58 [INFO] [Dynmap-Towny] Loading Dynmap-Towny v0.15.
12:44:58 [INFO] [Dynmap-Towny] initializing
12:44:58 [SEVERE] [Dynmap-Towny] Cannot find Towny!
12:44:58 [INFO] Server permissions file permissions.yml is empty, ignoring it
12:44:58 [INFO] Done (6.309s)! For help, type "help" or "?"

> # 

so like you see i'll got a error,, dont get it why this time, i look trough the issues in the hope to find someone the the same problem!, didnt find!,

a full review of my bPermissions:

default: Member
groups:
  globaladmin:
    permissions:
    - Mobhealth.command.*
    - Mobhealth.command.reload
    - Mobhealth.command.toggle
    - ^group.admin
    - alphachest.admin
    - alphachest.save
    - bPermissions.admin
    - bukkit.broadcast
    - bukkit.broadcast.admin
    - bukkit.command.give
    - bukkit.command.list
    - bukkit.command.xp
    - citizens.admin
    - citizens.admin.*
    - citizens.admin.override.*
    - citizens.basic.modify.*
    - citizens.blacksmith.*
    - citizens.evils.*
    - citizens.guard.*
    - citizens.guard.modify.*
    - citizens.guard.use.*
    - citizens.healer.*
    - citizens.healer.modify.*
    - citizens.npc.character
    - citizens.npc.create
    - citizens.npc.despawn
    - citizens.npc.equip
    - citizens.npc.help
    - citizens.npc.info
    - citizens.npc.lookclose
    - citizens.npc.path
    - citizens.npc.remove
    - citizens.npc.rename
    - citizens.npc.spawn
    - citizens.npc.text
    - citizens.npc.tp
    - citizens.npc.tphere
    - citizens.qaypoints.*
    - citizens.quester.*
    - citizens.quester.admin.*
    - citizens.quester.modify.*
    - citizens.toggle
    - citizens.trader.*
    - citizens.trader.admin
    - citizens.trader.modify.*
    - citizens.waypoints.modifier.*
    - citizens.wizard.*
    - citizens.wizard.modify.*
    - dynmap.add-id-for-ip
    - dynmap.cancelrender
    - dynmap.del-id-for-ip
    - dynmap.dmap.lightinglist
    - dynmap.dmap.mapadd
    - dynmap.dmap.mapdelete
    - dynmap.dmap.mapset
    - dynmap.dmap.perspectivelist
    - dynmap.dmap.shaderlist
    - dynmap.dmap.worldlist
    - dynmap.dmap.worldreset
    - dynmap.dmap.worldset
    - dynmap.fullrender
    - dynmap.hide.others
    - dynmap.hide.self
    - dynmap.ids-for-ip
    - dynmap.ips-for-id
    - dynmap.marker.add
    - dynmap.marker.addarea
    - dynmap.marker.addicon
    - dynmap.marker.addline
    - dynmap.marker.addset
    - dynmap.marker.delete
    - dynmap.marker.deletearea
    - dynmap.marker.deleteicon
    - dynmap.marker.deleteline
    - dynmap.marker.deleteset
    - dynmap.marker.icons
    - dynmap.marker.list
    - dynmap.marker.listareas
    - dynmap.marker.listlines
    - dynmap.marker.listsets
    - dynmap.marker.movehere
    - dynmap.marker.update
    - dynmap.marker.updatearea
    - dynmap.marker.updateicon
    - dynmap.marker.updateline
    - dynmap.marker.updateset
    - dynmap.purgequeue
    - dynmap.radiusrender
    - dynmap.reload
    - dynmap.render
    - dynmap.resetstats
    - dynmap.sendtoweb
    - dynmap.show.others
    - dynmap.show.self
    - dynmap.stats
    - dynmap.updaterender
    - gb.create
    - gb.delete
    - iConomy.*
    - iConomy.accounts.*
    - iConomy.accounts.create
    - iConomy.accounts.empty
    - iConomy.accounts.give
    - iConomy.accounts.purge
    - iConomy.accounts.remove
    - iConomy.accounts.set
    - iConomy.accounts.status
    - iConomy.accounts.status.set
    - iConomy.accounts.take
    - iConomy.help
    - iConomy.holdings.others
    - iConomy.plus
    - iConomy.user
    - jobs.admin.*
    - noafk.admin.config
    - towny.admin
    - towny.nation.*
    - towny.wild.*
    - towny.wild.switch
    - tracks.*
    groups:
    - admin
    - member
    - vip
    meta:
      priority: '100'
      prefix: '&6admin'
  Member:
    permissions:
    - Mobhealth.*
    - Mobhealth.show
    - bukkit.broadcast.user
    - citizens.*
    - citizens.basic.*
    - citizens.blacksmith.use.*
    - citizens.healer.use.*
    - citizens.npc.info
    - citizens.npc.select
    - citizens.quester.use.*
    - citizens.trader.use.*
    - citizens.wizard.use.*
    - gb.use
    - group.Member
    - iConomy.holdings
    - iConomy.payment
    - jobs.join.*
    - jobs.world.*
    - skillz.acrobatics
    - skillz.archery
    - skillz.axes
    - skillz.check
    - skillz.digging
    - skillz.farming
    - skillz.mining
    - skillz.skills
    - skillz.swimming
    - skillz.swords
    - skillz.unarmed
    - skillz.woodcutting
    - towny.claimed.*
    - towny.nation.*
    - towny.top
    - towny.town.*
    - towny.town.claim
    - towny.town.delete
    - towny.town.new
    - towny.town.plot
    - towny.town.resident
    - towny.town.spawn.*
    - towny.town.toggle.mobs
    - towny.town.toggle.public
    - towny.town.toggle.pvp
    - towny.wild.*
    - towny.wild.switch
    groups: []
    meta:
      prefix: ''
      suffix: ''

Please help me!,, im trying already a whole week too get it fully started!, youtube, Bukkit/citizens wiki, are help!

**Comment by Fox243** [comment]
Well if im not mistaken its going to be because bpermissions is updated its api to the new version so we have to wait for citizens to update there api to go with the new system. Thats just what iv gathered with reading around in bpermissions section 
bpermissions-v2.9 update
-THIS RELEASE WILL BREAK PLUGINS THAT RELY ON THE OLD BPERMISSIONS API
-THIS RELEASE WILL NO LONGER SUPPORT THE PERMISSIONS 2 BRIDGE
-THIS RELEASE WILL NOT WORK WITH PLUGINS THAT RELY ON THE OLD BPERMISSIONS API
-DO NOT UPDATE UNLESS YOU ARE SURE ALL YOURS PLUGINS ARE UP TO DATE
-THIS RELEASE MAY BREAK YOUR EXISTING PLUGINS
-THIS RELEASE REMOVES ALL LEGACY API SUPPORT
hope this helps 
also i have the same problem but i was forced to update bpermissions cause  it solves the problem with it resetting build mode 
for multi worlds
Sincerely Fox

**Comment by Ragehorror** [comment]
Dear fox,

the answer you gave, I am very grateful for. So i need to wait until the Citizens R5 + 2.0 comes out, i guess...

Sincrely Ragehorror

**Comment by fullwall** [comment]
You may get a fix for this when 1.2 minecraft comes out.
